xpk.py

```diff
@@ -3745,6 +4140,98 @@ def get_system_characteristics(args) -> tuple[SystemCharacteristics|None, int]:
   else:
     return None, 1
 
+
+def is_autoprovisioning_enabled(args, system: SystemCharacteristics) -> tuple[bool, int]:
+  """Determine if autoprovisioning is enabled.
+
+  Args:
+    args: user provided arguments for running the command.
+    system: system characteristics.
+
+  Returns:
+    bool is true if autoprovisioning is enabled, false otherwise.
+    int of 0 if successful and 1 otherwise.
+  """
+  resources_configmap_name = f'{args.cluster}-{_CLUSTER_RESOURCES_CONFIGMAP}'
+  cluster_config_map = get_cluster_configmap(args, resources_configmap_name)
+
+  if cluster_config_map is None:
+    xpk_print(f'Unable to find config map: {resources_configmap_name}. Autoprovisioning is not enabled.')
+    return False, 0
+
+  return_code, autoprovisioning_value = get_value_from_map(system.gke_accelerator, cluster_config_map)
+  if return_code != 0:
+    xpk_print(f'gke_accelerator type not found in config map: {resources_configmap_name}. Autoprovisioning is not enabled.')
+    return False, 0
+
+  if autoprovisioning_value == _AUTOPROVISIONING_CONFIG_VALUE:
+    xpk_print('Autoprovisioning is Enabled.')
+    return True, 0
+  else:
+    xpk_print(
+        'Error: Autoprovisioning not enabled but should be so exiting xpk. Value should be'
+        f' {_AUTOPROVISIONING_CONFIG_VALUE} but instead found value of '
+        f' {cluster_config_map[system.accelerator_type]}'
+    )
+    return False, 1
+
+
+def get_autoprovisioning_node_selector_args(args) -> tuple[str, int]:
+  """Determine the capacity type when autoprovisioning is enabled.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    Tuple with string of autoprovisioning node selector args and
+    int of 0 if successful and 1 otherwise.
+  """
+  return_code = 0
+  node_selector_args = ''
+  # If the user doesn't specify args, then use the cluster settings.
+  capacity_type, return_code = get_capacity_type(args)
+  capacity_type_str = capacity_type.name
+  if return_code != 0:
+    xpk_print('Unable to get capacity type.')
+    return node_selector_args, return_code
+
+  if capacity_type_str == CapacityType.UNKNOWN.name:
+    # Use default settings from cluster creation.
+    resources_configmap_name = f'{args.cluster}-{_CLUSTER_METADATA_CONFIGMAP}'
```

Comment:
nit: `metadata_configmap_name` to avoid confusion

Reply:
agreed